src/index.ts

```diff
@@ -0,0 +1,119 @@
+    AutojoinRoomsMixin.setupOnClient(client);
+
+    // Ensure we're joined to the safety team room
+    await client.joinRoom(safetyTeamRoomId);
+
+    client.on("room.join", async (roomId: string, event: MatrixEvent) => {
+        console.log(`Joined ${roomId}`);
+        await client.sendHtmlNotice(roomId, "Hello! To get started using policyserv, say <code>!policyserv community &lt;community name&gt;</code>. For more information, say <code>!policyserv help</code>.<br/><br/><b>Note</b>: This bot will be replaced with a web interface in the future. Monitor the <a href='https://matrix.org/blog/'>matrix.org blog</a> for updates.");
+    });
+
+    client.on("room.message", async (roomId: string, event: any) => {
+        if (event.sender === await client.getUserId()) {
+            return; // ignore ourselves
+        }
+        if (event.type !== "m.room.message") {
+            return;
+        }
+        const textEvent = new MatrixEvent<TextualMessageEventContent>(event);
+        if (textEvent.content.msgtype !== "m.text" || !textEvent.content.body) {
+            return;
+        }
+
+        if (textEvent.content.body.toLowerCase().startsWith("!policyserv ")) {
+            const args = textEvent.content.body.substring("!policyserv ".length).split(" ");
+            if (args[0] === "help") {
+                await client.replyHtmlNotice(roomId, event,
+                    "This bot is used to manage a community's policyserv settings. <br/>" +
+                    "All commands require a community to exist first. To do so, say <code>!policyserv community &lt;community name&gt;</code>.<br/>" +
+                    "Afterwards, the following commands will be available:<br/>" +
+                    "TODO" // TODO: Specify commands as they exist in future PRs
+                );
+            } else if (args[0] === "community") {
+                if (!!storageProvider.readValue(`room:${roomId}`)) {
+                    await client.replyHtmlNotice(roomId, event, "❌ This room is already associated with a community.");
+                    return;
+                }
+
+                if (args.length < 2) {
+                    await client.replyHtmlNotice(roomId, event, "Please specify a community name.");
+                    return
+                }
+                const communityName = args.slice(1).join(" ");
+
+                // Give some indication that the bot saw the command ahead of doing the work
+                const reactionEventId = await client.unstableApis.addReactionToEvent(roomId, event.event_id, "👀");
+                try {
+                    const communityId = await policyservApi.createCommunity(communityName);
+                    storageProvider.storeValue(`room:${roomId}`, JSON.stringify({id: communityId}));
+                    // noinspection ES6MissingAwait - we aren't concerned if this fails
+                    client.redactEvent(roomId, reactionEventId);
```

Comment:
just curious why the redaction here

Reply:
mostly just to acknowledge that the command was actioned. Mjolnir does a similar pattern on some commands:

1. Add a reaction to show the user it's been seen
2. Do the long-running action (which in this case should be pretty quick, but it's still a network request)
3. Remove and/or replace the reaction to indicate that the command was finished

Mjolnir uses `Pending` with a checkmark after. Here we're just using eyes with removal instead.